pkg/sdk/roles.go

```diff
@@ -0,0 +1,268 @@
+	if err := json.Unmarshal(body, &res); err != nil {
+		return []string{}, errors.NewSDKError(err)
+	}
+
+	return res.Members, nil
+}
+
+func (sdk mgSDK) listRoleMembers(entityURL, entityEndpoint, id, roleName, domainID string, pm PageMetadata, token string) (RoleMembersPage, errors.SDKError) {
+	endpoint := fmt.Sprintf("%s/%s/%s/%s/%s/%s", domainID, entityEndpoint, id, rolesEndpoint, roleName, membersEndpoint)
+	if entityEndpoint == domainsEndpoint {
+		endpoint = fmt.Sprintf("%s/%s/%s/%s/%s", entityEndpoint, id, rolesEndpoint, roleName, membersEndpoint)
+	}
+	url, err := sdk.withQueryParams(entityURL, endpoint, pm)
+	if err != nil {
+		return RoleMembersPage{}, errors.NewSDKError(err)
+	}
+	_, body, sdkerr := sdk.processRequest(http.MethodGet, url, token, nil, nil, http.StatusOK)
+	if sdkerr != nil {
+		return RoleMembersPage{}, sdkerr
+	}
+
+	res := RoleMembersPage{}
+	if err := json.Unmarshal(body, &res); err != nil {
+		return RoleMembersPage{}, errors.NewSDKError(err)
+	}
+
+	return res, nil
+}
+
+func (sdk mgSDK) removeRoleMembers(entityURL, entityEndpoint, id, roleName, domainID string, members []string, token string) errors.SDKError {
+	rcrm := roleMembersReq{Members: members}
+	data, err := json.Marshal(rcrm)
+	if err != nil {
+		return errors.NewSDKError(err)
+	}
+
+	url := fmt.Sprintf("%s/%s/%s/%s/%s/%s/%s/%s", entityURL, domainID, entityEndpoint, id, rolesEndpoint, roleName, membersEndpoint, "delete")
+	if entityEndpoint == domainsEndpoint {
+		url = fmt.Sprintf("%s/%s/%s/%s/%s/%s/%s", entityURL, entityEndpoint, id, rolesEndpoint, roleName, membersEndpoint, "delete")
+	}
+	_, _, sdkerr := sdk.processRequest(http.MethodPost, url, token, data, nil, http.StatusNoContent)
+
+	return sdkerr
+}
+
+func (sdk mgSDK) removeAllRoleMembers(entityURL, entityEndpoint, id, roleName, domainID, token string) errors.SDKError {
+	url := fmt.Sprintf("%s/%s/%s/%s/%s/%s/%s/%s", entityURL, domainID, entityEndpoint, id, rolesEndpoint, roleName, membersEndpoint, "delete-all")
+	if entityEndpoint == domainsEndpoint {
+		url = fmt.Sprintf("%s/%s/%s/%s/%s/%s/%s", entityURL, entityEndpoint, id, rolesEndpoint, roleName, membersEndpoint, "delete-all")
+	}
+	_, _, sdkerr := sdk.processRequest(http.MethodPost, url, token, nil, nil, http.StatusNoContent)
+
+	return sdkerr
+}
+
+func (sdk mgSDK) listAvailableRoleActions(entityURL, entityEndpoint, domainID, token string) ([]string, errors.SDKError) {
+	url := fmt.Sprintf("%s/%s/%s/%s/%s", entityURL, domainID, entityEndpoint, rolesEndpoint, "available-actions")
+	if entityEndpoint == domainsEndpoint {
+		url = fmt.Sprintf("%s/%s/%s/%s", entityURL, entityEndpoint, rolesEndpoint, "available-actions")
+	}
+	_, body, sdkerr := sdk.processRequest(http.MethodGet, url, token, nil, nil, http.StatusOK)
+	if sdkerr != nil {
+		return nil, sdkerr
+	}
+
+	res := availableRoleActionsRes{}
+	if err := json.Unmarshal(body, &res); err != nil {
+		return nil, errors.NewSDKError(err)
+	}
+
+	return res.AvailableActions, nil
+}
```

Comment:
For Domains, we are passing empty `domainID` i.e empty string `""` , if `domainID` is empty will create `//` in request url , Does it works correctly ?

Reply:
There is a check for incase it is for domains, to change the format of the url
`
if entityEndpoint == domainsEndpoint {
		url = fmt.Sprintf("%s/%s/%s/%s", entityURL, entityEndpoint, rolesEndpoint, "available-actions")
	}
`